dask/dataframe/io/csv.py

```diff
@@ -68,6 +71,8 @@ def pandas_read_text(reader, b, header, kwargs, dtypes=None, columns=None,
         raise ValueError("Columns do not match", df.columns, columns)
     elif columns:
         df.columns = columns
+    if path:
+        df = df.assign(**path)
```

Comment:
We should take some care to not clobber an existing column named `path` here.

Typically pandas will append `.1`, `.2`, etc. to column names if there are duplicates. IIRC this is controlled by `mangle_dupe_cols`.

Reply:
Right but since we are doing this after the `read_csv`, I don't we really need `mangle_dupe_cols`, we probably just need to check for `path` in `df.columns` and append .n to it until we find something that isn't in `df.columns` or throw an error and tell the user to specify the `include_path_column` name.